rtabmap nodes crash immediately in ROS 2 Eloquent w/ "Failed to find library 'rtabmap_ros__rosidl_typesupport_introspection_cpp'"

OS: Ubuntu Linux 18.04
Distro: ROS Eloquent

I know ROS 2 support is still experimental, but I feel like the nodes should at least start without crashing.  After building the `rtabmap_ros` package and trying to launch one of the default launch files with `ros2 launch rtabmap_ros turtlebot3_scan.launch.py`, both nodes throw an exception and die like so:
```bash
$ ros2 launch rtabmap_ros turtlebot3_scan.launch.py 
[INFO] [launch]: All log files can be found below /home/preed/.ros/log/2020-03-27-23-30-45-844130-rover-nuc-26382
[INFO] [launch]: Default logging verbosity is set to INFO
1585369845.847427 [0]       ros2: using network interface eno1 (udp/192.168.2.100) selected arbitrarily from: eno1, docker0
[INFO] [rtabmap-1]: process started with pid [26401]
[INFO] [rtabmapviz-2]: process started with pid [26402]
[rtabmap-1] 1585369846.089611 [0]    rtabmap: using network interface eno1 (udp/192.168.2.100) selected arbitrarily from: eno1, docker0
[rtabmap-1] [INFO] [rtabmap]: rtabmap: subscribe_depth = false
[rtabmap-1] [INFO] [rtabmap]: rtabmap: subscribe_rgb = false
[rtabmap-1] [INFO] [rtabmap]: rtabmap: subscribe_stereo = false
[rtabmap-1] [INFO] [rtabmap]: rtabmap: subscribe_rgbd = false (rgbd_cameras=1)
[rtabmap-1] [INFO] [rtabmap]: rtabmap: subscribe_odom_info = false
[rtabmap-1] [INFO] [rtabmap]: rtabmap: subscribe_user_data = false
[rtabmap-1] [INFO] [rtabmap]: rtabmap: subscribe_scan = true
[rtabmap-1] [INFO] [rtabmap]: rtabmap: subscribe_scan_cloud = false
[rtabmap-1] [INFO] [rtabmap]: rtabmap: queue_size    = 10
[rtabmap-1] [INFO] [rtabmap]: rtabmap: approx_sync   = true
[rtabmap-1] [INFO] [rtabmap]: rtabmap(maps): map_filter_radius          = 0.000000
[rtabmap-1] [INFO] [rtabmap]: rtabmap(maps): map_filter_angle           = 30.000000
[rtabmap-1] [INFO] [rtabmap]: rtabmap(maps): map_cleanup                = true
[rtabmap-1] [INFO] [rtabmap]: rtabmap(maps): map_always_update          = false
[rtabmap-1] [INFO] [rtabmap]: rtabmap(maps): map_empty_ray_tracing      = true
[rtabmap-1] [INFO] [rtabmap]: rtabmap(maps): cloud_output_voxelized     = true
[rtabmap-1] [INFO] [rtabmap]: rtabmap(maps): cloud_subtract_filtering   = false
[rtabmap-1] [INFO] [rtabmap]: rtabmap(maps): cloud_subtract_filtering_min_neighbors = 2
[rtabmap-1] [INFO] [rtabmap]: rtabmap: frame_id      = base_footprint
[rtabmap-1] [INFO] [rtabmap]: rtabmap: map_frame_id  = map
[rtabmap-1] [INFO] [rtabmap]: rtabmap: use_action_for_goal  = false
[rtabmap-1] [INFO] [rtabmap]: rtabmap: tf_delay      = 0.050000
[rtabmap-1] [INFO] [rtabmap]: rtabmap: tf_tolerance  = 0.100000
[rtabmap-1] [INFO] [rtabmap]: rtabmap: odom_sensor_sync   = false
[rtabmap-1] Failed to find library 'rtabmap_ros__rosidl_typesupport_introspection_cpp'
[rtabmap-1] 
[rtabmap-1] >>> [rcutils|error_handling.c:107] rcutils_set_error_state()
[rtabmap-1] This error state is being overwritten:
[rtabmap-1] 
[rtabmap-1]   'type support not from this implementation, at /tmp/binarydeb/ros-eloquent-rmw-cyclonedds-cpp-0.4.4/src/rmw_node.cpp:991'
[rtabmap-1] 
[rtabmap-1] with this new error message:
[rtabmap-1] 
[rtabmap-1]   'type_support is null, at /tmp/binarydeb/ros-eloquent-rmw-cyclonedds-cpp-0.4.4/src/rmw_node.cpp:1212'
[rtabmap-1] 
[rtabmap-1] rcutils_reset_error() should be called after error handling to avoid this.
[rtabmap-1] <<<
[rtabmap-1] terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
[rtabmap-1]   what():  could not create publisher: type_support is null, at /tmp/binarydeb/ros-eloquent-rmw-cyclonedds-cpp-0.4.4/src/rmw_node.cpp:1212, at /tmp/binarydeb/ros-eloquent-rcl-0.8.4/src/rcl/publisher.c:173
[ERROR] [rtabmap-1]: process has died [pid 26401, exit code -6, cmd '/home/preed/src/ros2/rtabmap/install/rtabmap_ros/lib/rtabmap_ros/rtabmap -d --ros-args --params-file /tmp/launch_params_n8tzjyd5 -r scan:=/scan'].
[rtabmapviz-2] 1585369846.227603 [0] rtabmapviz: using network interface eno1 (udp/192.168.2.100) selected arbitrarily from: eno1, docker0
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: subscribe_depth = false
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: subscribe_rgb = false
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: subscribe_stereo = false
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: subscribe_rgbd = false (rgbd_cameras=1)
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: subscribe_odom_info = false
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: subscribe_user_data = false
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: subscribe_scan = true
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: subscribe_scan_cloud = false
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: queue_size    = 10
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: approx_sync   = true
[rtabmapviz-2] [INFO] [rtabmapviz]: rtabmapviz: Using configuration from "/home/preed/.ros/rtabmapGUI.ini"
[rtabmapviz-2] libpng warning: iCCP: known incorrect sRGB profile
[rtabmapviz-2] libpng warning: iCCP: known incorrect sRGB profile
[rtabmapviz-2] libpng warning: iCCP: known incorrect sRGB profile
[rtabmapviz-2] [WARN] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[rtabmapviz-2] [INFO] [rtabmapviz]: Reading parameters from the ROS server...
[rtabmapviz-2] [ERROR] [rtabmapviz]: Can't call rtabmap parameters service, is the node running?
[rtabmapviz-2] [INFO] [rtabmapviz]: Parameters read = 0
[rtabmapviz-2] Failed to find library 'rtabmap_ros__rosidl_typesupport_introspection_cpp'
[rtabmapviz-2] 
[rtabmapviz-2] >>> [rcutils|error_handling.c:107] rcutils_set_error_state()
[rtabmapviz-2] This error state is being overwritten:
[rtabmapviz-2] 
[rtabmapviz-2]   'type support not from this implementation, at /tmp/binarydeb/ros-eloquent-rmw-cyclonedds-cpp-0.4.4/src/rmw_node.cpp:991'
[rtabmapviz-2] 
[rtabmapviz-2] with this new error message:
[rtabmapviz-2] 
[rtabmapviz-2]   'type_support is null, at /tmp/binarydeb/ros-eloquent-rmw-cyclonedds-cpp-0.4.4/src/rmw_node.cpp:1433'
[rtabmapviz-2] 
[rtabmapviz-2] rcutils_reset_error() should be called after error handling to avoid this.
[rtabmapviz-2] <<<
[rtabmapviz-2] terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
[rtabmapviz-2]   what():  could not create subscription: type_support is null, at /tmp/binarydeb/ros-eloquent-rmw-cyclonedds-cpp-0.4.4/src/rmw_node.cpp:1433, at /tmp/binarydeb/ros-eloquent-rcl-0.8.4/src/rcl/subscription.c:168
[ERROR] [rtabmapviz-2]: process has died [pid 26402, exit code -6, cmd '/home/preed/src/ros2/rtabmap/install/rtabmap_ros/lib/rtabmap_ros/rtabmapviz --ros-args --params-file /tmp/launch_params_7ln5gpq9 -r scan:=/scan'].
```

I'm using `rmw_cyclonedds_cpp` as my default DDS implementation, but `fastrtps` and `opensplice` both fail with similar errors.  I can also verify that, like the error message says, no library named `rtabmap_ros__rosidl_typesupport_introspection_cpp` was built inside either my `build` or `install` directory.  Am I missing something?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -117,10 +117,11 @@ find_package(rviz_default_plugins REQUIRED)
  )
 
 ## Generate messages and services
- rosidl_generate_interfaces(rtabmap_msgs
+ rosidl_generate_interfaces(${PROJECT_NAME}_msgs
    ${msg_files}
    ${srv_files}
    DEPENDENCIES geometry_msgs std_msgs sensor_msgs std_srvs visualization_msgs image_geometry
+   LIBRARY_NAME ${PROJECT_NAME}
  )
 ament_export_dependencies(rosidl_default_runtime)
 
@@ -270,9 +271,9 @@ target_link_libraries(rtabmap_ros ${RTABMap_LIBRARIES})
 target_link_libraries(rtabmap_sync rtabmap_ros ${RTABMap_LIBRARIES})
 target_link_libraries(rtabmap_plugins rtabmap_ros ${RTABMap_LIBRARIES})
 
-rosidl_target_interfaces(rtabmap_ros rtabmap_msgs "rosidl_typesupport_cpp")
-rosidl_target_interfaces(rtabmap_sync rtabmap_msgs "rosidl_typesupport_cpp")
-rosidl_target_interfaces(rtabmap_plugins rtabmap_msgs "rosidl_typesupport_cpp")
+rosidl_target_interfaces(rtabmap_ros ${PROJECT_NAME}_msgs "rosidl_typesupport_cpp")
+rosidl_target_interfaces(rtabmap_sync ${PROJECT_NAME}_msgs "rosidl_typesupport_cpp")
+rosidl_target_interfaces(rtabmap_plugins ${PROJECT_NAME}_msgs "rosidl_typesupport_cpp")
 
 rclcpp_components_register_nodes(rtabmap_plugins "rtabmap_ros::RGBDOdometry")
 rclcpp_components_register_nodes(rtabmap_plugins "rtabmap_ros::StereoOdometry")
@@ -421,34 +422,34 @@ get_rmw_typesupport(typesupport_impls "${rmw_implementation}" LANGUAGE "cpp")
 
 foreach(typesupport_impl ${typesupport_impls})
   rosidl_target_interfaces(rtabmap_rgbd_odometry
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
   rosidl_target_interfaces(rtabmap_stereo_odometry
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
   rosidl_target_interfaces(rtabmap_icp_odometry
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
   rosidl_target_interfaces(rtabmap_rgbd_relay
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
   rosidl_target_interfaces(rtabmap_rgbd_sync
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
   rosidl_target_interfaces(rtabmap_stereo_sync
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
   rosidl_target_interfaces(rtabmap_pointcloud_to_depthimage
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
   rosidl_target_interfaces(rtabmap_point_cloud_xyzrgb
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
   rosidl_target_interfaces(rtabmap
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
   rosidl_target_interfaces(rtabmapviz
-    rtabmap_msgs ${typesupport_impl}
+    ${PROJECT_NAME}_msgs ${typesupport_impl}
   )
 endforeach()
 
@@ -507,7 +508,7 @@ IF(rviz_default_plugins_FOUND)
 
     foreach(typesupport_impl ${typesupport_impls})
 	  rosidl_target_interfaces(rtabmap_rviz_plugins
-	    rtabmap_msgs ${typesupport_impl}
+	    ${PROJECT_NAME}_msgs ${typesupport_impl}
 	  )
     endforeach()
     

--- a/package.xml
+++ b/package.xml
@@ -53,6 +53,7 @@
   <exec_depend>std_msgs</exec_depend>
   <exec_depend>std_srvs</exec_depend>
   <exec_depend>nav_msgs</exec_depend>
+  <exec_depend>nav2_common</exec_depend>
   <exec_depend>stereo_msgs</exec_depend>
   <exec_depend>geometry_msgs</exec_depend>
   <exec_depend>visualization_msgs</exec_depend>
@@ -70,7 +71,6 @@
   <exec_depend>octomap</exec_depend>
   <exec_depend>image_geometry</exec_depend>
   <exec_depend>find_object_2d</exec_depend>
-  <exec_depend>message_runtime</exec_depend>
   <exec_depend>pluginlib</exec_depend>
 
   <build_depend>libpcl-all-dev</build_depend>

